src/App.tsx

```diff
@@ -9,39 +9,27 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { clearPost, userPosts } from './features/posts/postsSlice';
+import { clearSelectedPost } from './features/selectedPost/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  const dispatch = useAppDispatch();
+  const { posts, loaded, hasError } = useAppSelector(state => state.userPost);
+  const selectedPost = useAppSelector(state => state.selectedPost.selectedPost);
+  const { author } = useAppSelector(state => state.author);
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(clearPost());
+    dispatch(clearSelectedPost());
```

Comment:
The `clearPost` and `clearSelectedPost` actions are dispatched twice: once at the beginning of the `useEffect` and again in the `else` block. This redundancy might not be necessary unless there is a specific reason for clearing the state twice. Consider removing the initial dispatches if they are not needed.